vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/openid/RequestOptions.kt

```diff
@@ -0,0 +1,236 @@
+package at.asitplus.wallet.lib.openid
+
+import at.asitplus.jsonpath.core.NormalizedJsonPath
+import at.asitplus.jsonpath.core.NormalizedJsonPathSegment
+import at.asitplus.wallet.lib.data.AttributeIndex
+import at.asitplus.wallet.lib.data.ConstantIndex
+import at.asitplus.wallet.lib.data.ConstantIndex.CredentialRepresentation
+import at.asitplus.wallet.lib.data.ConstantIndex.CredentialRepresentation.ISO_MDOC
+import at.asitplus.wallet.lib.data.ConstantIndex.CredentialRepresentation.PLAIN_JWT
+import at.asitplus.wallet.lib.data.ConstantIndex.CredentialRepresentation.SD_JWT
+import at.asitplus.wallet.lib.data.ConstantIndex.supportsIso
+import at.asitplus.wallet.lib.data.ConstantIndex.supportsSdJwt
+import at.asitplus.wallet.lib.data.ConstantIndex.supportsVcJwt
+import at.asitplus.wallet.lib.data.VcDataModelConstants.VERIFIABLE_CREDENTIAL
+import at.asitplus.wallet.lib.data.dif.Constraint
+import at.asitplus.wallet.lib.data.dif.ConstraintField
+import at.asitplus.wallet.lib.data.dif.ConstraintFilter
+import at.asitplus.wallet.lib.data.dif.InputDescriptor
+import at.asitplus.wallet.lib.data.dif.SchemaReference
+import at.asitplus.wallet.lib.oidc.OidcSiopVerifier.AuthnResponseResult
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import at.asitplus.wallet.lib.oidc.OpenIdConstants.CREDENTIAL_TYPE_OPENID
+import at.asitplus.wallet.lib.oidc.OpenIdConstants.SCOPE_OPENID
+import at.asitplus.wallet.lib.oidc.OpenIdConstants.SCOPE_PROFILE
+import at.asitplus.wallet.lib.oidvci.AuthorizationDetails
+import at.asitplus.wallet.lib.oidvci.CredentialFormatEnum
+import at.asitplus.wallet.lib.oidvci.CredentialRequestParameters
+import at.asitplus.wallet.lib.oidvci.CredentialRequestProof
+import at.asitplus.wallet.lib.oidvci.SupportedCredentialFormatDefinition
+import at.asitplus.wallet.lib.oidvci.toRequestedClaimsIso
+import at.asitplus.wallet.lib.oidvci.toRequestedClaimsSdJwt
+import com.benasher44.uuid.uuid4
+import io.ktor.http.*
+import kotlinx.datetime.Clock
+
+
+data class RequestOptions(
```

Comment:
But now not all parameters here make sense for both, SIOPv2 and OID4VCI, right?

Reply:
OID4VCI does not use `responseMode`, `clientMetadataUrl` and `encryption`, however by their default values they do not interfere with this use case (nullable) and they are ignored for all OID4VCI use-cases.
The same goes for SIOPv2 and clock. Since we never use RequestOptions as object but rather access it's members directly this does not influence the usage. Additionally we can make clock transient for serialization to make sure that it is never leaked.

Reply:
Should i add this to the variable documentation?